docs/adding-a-provider.md

```diff
@@ -0,0 +1,106 @@
+# Adding a Provider
+
+This is a brief guide explaining what needs to be done to get Halyard to
```

Comment:
The end of this sentence sounds a little off to me. Maybe "This is a brief guide explaining what needs to be done to add support to Halyard for an additional cloud provider."?